homeassistant/components/hue/sensor_base.py

```diff
@@ -0,0 +1,274 @@
+        _LOGGER.info('Starting sensor polling loop with %s second interval',
+                     self.SCAN_INTERVAL.total_seconds())
+
+        async def async_update_bridge(now):
+            """Will update sensors from the bridge."""
+            await self.async_update_items()
+
+            async_track_point_in_utc_time(
+                self.hass, async_update_bridge, utcnow() + self.SCAN_INTERVAL)
+
+        await async_update_bridge(None)
+
+    async def async_update_items(self):
+        """Update sensors from the bridge."""
+        import aiohue
+
+        api = self.bridge.api.sensors
+
+        try:
+            start = monotonic()
+            with async_timeout.timeout(4):
+                await api.update()
+        except (asyncio.TimeoutError, aiohue.AiohueException) as err:
+            _LOGGER.debug('Failed to fetch sensor: %s', err)
+
+            if not self.bridge.available:
+                return
+
+            _LOGGER.error('Unable to reach bridge %s (%s)', self.bridge.host,
+                          err)
+            self.bridge.available = False
+
+            return
+
+        finally:
+            _LOGGER.debug('Finished sensor request in %.3f seconds',
+                          monotonic() - start)
+
+        if not self.bridge.available:
+            _LOGGER.info('Reconnected to bridge %s', self.bridge.host)
+            self.bridge.available = True
+
+        new_sensors = []
+        new_binary_sensors = []
+        primary_sensor_devices = {}
+        current = self.hass.data[hue.DOMAIN][CURRENT_SENSORS]
+
+        # Physical Hue motion sensors present as three sensors in the API: a
+        # presence sensor, a temperature sensor, and a light level sensor. Of
+        # these, only the presence sensor is assigned the user-friendly name
+        # that the user has given to the device. Each of these sensors is
+        # linked by a common device_id, which is the first twenty-three
+        # characters of the unique id (then followed by a hyphen and an ID
+        # specific to the individual sensor).
+        #
+        # To set up neat values, and assign the sensor entities to the same
+        # device, we first, iterate over all the sensors and find the Hue
+        # presence sensors, then iterate over all the remaining sensors -
+        # finding the remaining ones that may or may not be related to the
+        # presence sensors.
+        for item_id in api:
+            if api[item_id].type != aiohue.sensors.TYPE_ZLL_PRESENCE:
+                continue
+
+            primary_sensor_devices[_device_id(api[item_id])] = api[item_id]
+
+        # Iterate again now we have all the presence sensors, and add the
+        # related sensors with nice names where appropriate.
+        for item_id in api:
+            existing = current.get(api[item_id].uniqueid)
+            if existing is not None:
+                if existing.hass is not None:
+                    existing.async_schedule_update_ha_state()
```

Comment:
Use our dispatch helper instead to signal the entity to update state. That will make sure we don't ask it to update before the entity has been added to home assistant.